tensorflow/compiler/jit/deadness_analysis.cc

```diff
@@ -937,8 +1092,21 @@ Status DeadnessAnalysisImpl::HandleMerge(Node* n,
       // We're visiting this merge for the first time and it has an unvisited
       // backedge.
       Predicate* input_data_pred;
-      TF_RETURN_IF_ERROR(predicate_factory_.MakeSymbolPredicate(
-          n, /*output_idx=*/0, /*must_be_true=*/false, &input_data_pred));
+      if (use_optimistic_mode) {
+        // In the optimistic mode, we use the first-seen Merge node per
+        // frame as the representative Merge node.  It is just convenient and
+        // does not affect the result after pattern-matching into the
+        // AndRecurrence form.
+        string frame_name = control_flow_info_[n->id()].frame_name;
+        auto ret = frame_to_merge_node_.insert({frame_name, n});
```

Comment:
Maybe s/`ret`/`insert_result`/ ?  Or you could avoid the temporary object altogether and do

`Node* representative = frame_to_merge_node_.insert({frame_name, n}).first->second;`